server/integration/jgroups/src/main/java/org/infinispan/server/jgroups/ManagedSocketFactory.java

```diff
@@ -80,7 +80,7 @@ private Socket register(final Socket socket) {
 
     @Override
     public ServerSocket createServerSocket(String name) throws IOException {
-        return this.register(this.factory.createServerSocket(name));
+        return this.factory.createServerSocket(name);
```

Comment:
Found the WildFly bug: https://issues.jboss.org/browse/WFCORE-1127
It's marked as fixed in 2.0.2, but `SocketBindingManager.UnnamedBindingRegistryImpl` still throws the `IllegalStateException` in wildfly-network 2.2.0.Final.